lib/lte_link_control/modules/dns.c

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stddef.h>
+#include <zephyr/net/socket.h>
+#include <nrf_socket.h>
+
+LOG_MODULE_DECLARE(lte_lc, CONFIG_LTE_LINK_CONTROL_LOG_LEVEL);
+
+/* The DNS address is just a string, we don't know if it's IPv4 or IPv6 address.
+ * Check with inet_pton() to determine the right address family.
+ */
+void dns_fallback_set(void)
+{
+	int err;
+	int success;
+	struct nrf_in_addr in_addr;
+	struct nrf_in6_addr in6_addr;
+
+	success = nrf_inet_pton(NRF_AF_INET, CONFIG_LTE_LC_DNS_FALLBACK_ADDRESS, &in_addr);
+	if (success) {
+		err = nrf_setdnsaddr(NRF_AF_INET, &in_addr, sizeof(in_addr));
+		if (err) {
+			LOG_ERR("Failed to set fallback DNS address, err %d", errno);
+			return;
+		}
+	}
+
+	success = nrf_inet_pton(NRF_AF_INET6, CONFIG_LTE_LC_DNS_FALLBACK_ADDRESS, &in6_addr);
```

Comment:
The implementation currently always tries both IPv4 and IPv6. I think it would be nicer to skip the IPv6 check if the address family was already determined to be IPv4.

Reply:
Updated!